crypto/vrf/vrf.go

```diff
@@ -60,15 +59,15 @@ func GenerateKey(rnd io.Reader) (pk *PublicKey, sk *PrivateKey, err error) {
 	pkP.ToBytes(&pkBytes)
 
 	copy(sk[32:], pkBytes[:])
-	copy(pk[:], pkBytes[:])
-	return pk, sk, err
+	return
 }
 
-func (sk *PrivateKey) Public() []byte {
-	return ed25519.PrivateKey(sk[:]).Public().(ed25519.PublicKey)
+func (sk *PrivateKey) Public() (PublicKey, bool) {
+	pk, ok := ed25519.PrivateKey(*sk).Public().(ed25519.PublicKey)
+	return PublicKey(pk), ok
 }
 
-func (sk *PrivateKey) expandSecret() (x, skhr *[32]byte) {
+func (sk PrivateKey) expandSecret() (x, skhr *[32]byte) {
```

Comment:
I think, we should also align this in sign and vrf packages, too. As explained above pointers bring no benefit here and I would prefer values but pointers won't harm and have no real downside (some extra `*` here and there).